x-pack/plugins/osquery/public/plugin.ts

```diff
@@ -95,6 +95,7 @@ export class OsqueryPlugin implements Plugin<OsqueryPluginSetup, OsqueryPluginSt
         storage: this.storage,
         kibanaVersion: this.kibanaVersion,
       }),
+      osqueryMenuItem: useOsqueryMenuItem,
```

Comment:
I'm not sure if this method will scale, Osquery plugin should not be aware of any implementation details of other plugins, so in my opinion, this method should be just returning the Boolean to indicate if Osquery is available on a particular agent/host

Reply:
example of Lens `canUseEditor` method
https://github.com/elastic/kibana/blob/main/x-pack/plugins/lens/public/plugin.ts#L427

Reply:
Good point! 👍